.github/workflows/turing.yaml

```diff
@@ -167,23 +167,17 @@ jobs:
         working-directory: ui
     steps:
       - name: Check out code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       - name: Install system dependencies
         run: sudo apt-get install --no-install-recommends gcc make libpng-dev
 
-      - name: Set up Node 16.x
-        uses: actions/setup-node@v1
+      - name: Set up Node 20.x
+        uses: actions/setup-node@v4
         with:
-          node-version: "16.x"
-
-      - name: Cache Dependencies
-        uses: actions/cache@v2
-        with:
-          path: ui/node_modules
-          key: |
-            ${{ runner.os }}-modules-${{ hashFiles('ui/yarn.lock') }}
-          restore-keys: ${{ runner.os }}-modules-
+          node-version: "20.x"
+          cache: yarn
+          cache-dependency-path: ui/yarn.lock
```

Comment:
Example of the cache being restored successfully: https://github.com/caraml-dev/turing/actions/runs/9804032065/job/27071120656#step:4:36